### PR TITLE
All: catch exception and don't compute co2 when there is no solution

### PR DESCRIPTION
**Pull request by xlqian**
- catch exception 
- don't compute co2 when there is no solution

### CODE REVIEW

**Comment by xlqian** [comment]
I assume that catch the exception is a little gross but at least, it's not too difficult to understand... I've thought about returning an optional value instead of catching exception, but with `navitia::JourneysArg arg` it'll be complicated

File: source/kraken/worker.cpp [hunk 1]
```diff
@@ -85,6 +85,8 @@ static type::GeographicalCoord coord_of_entry_point(
             return poi->second->coord;
         }
     }
+    LOG4CPLUS_ERROR(log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("logger")),
```
Comment (kinnou02) [inline source/kraken/worker.cpp]: I don't want error log, it's not an error of the application!

File: source/kraken/worker.cpp [hunk 2]
```diff
@@ -85,6 +85,8 @@ static type::GeographicalCoord coord_of_entry_point(
...
+            "The entry point: " + entry_point.uri + " is not valid");
```
Comment (antoine-de) [inline source/kraken/worker.cpp]: minor: it's better to always use the `<<`

File: source/kraken/worker.cpp [hunk 3]
```diff
@@ -415,7 +417,14 @@ pbnavitia::Response Worker::proximity_list(const pbnavitia::PlacesNearbyRequest
                                            const boost::posix_time::ptime& current_datetime) {
     const auto data = data_manager.get_data();
     type::EntryPoint ep(data->get_type_of_id(request.uri()), request.uri());
-    auto coord = coord_of_entry_point(ep, data);
+    type::GeographicalCoord  coord;
+    try{
+        coord = coord_of_entry_point(ep, data);
+    }catch(const std::exception& e) {
```
Comment (TeXitoi) [inline source/kraken/worker.cpp]: catching such an anonymous exception will cause problem. I'm not fond of this catch everywhere, but at least it's cleaner with a dedicated exception.

File: source/kraken/worker.cpp (hunk 1 above)
Comment (xlqian) [inline source/kraken/worker.cpp]: OK, changed to a debug log

File: source/kraken/worker.cpp (hunk 2 above)
Comment (xlqian) [inline source/kraken/worker.cpp]: thx

File: source/kraken/worker.cpp (hunk 3 above)
Comment (xlqian) [inline source/kraken/worker.cpp]: done

**Comment by xlqian** [comment]
Jenkins, test this please!

File: source/kraken/worker.cpp [hunk 4]
```diff
@@ -415,7 +417,14 @@ pbnavitia::Response Worker::proximity_list(const pbnavitia::PlacesNearbyRequest
                                            const boost::posix_time::ptime& current_datetime) {
     const auto data = data_manager.get_data();
     type::EntryPoint ep(data->get_type_of_id(request.uri()), request.uri());
-    auto coord = coord_of_entry_point(ep, data);
+    type::GeographicalCoord  coord;
+    try{
+        coord = coord_of_entry_point(ep, data);
+    }catch(const navitia::recoverable_exception& e) {
```
Comment (TeXitoi) [inline source/kraken/worker.cpp]: `recoverable_exception` can be something else. You'll need a dedicated type.

**Comment by TeXitoi** [comment]
OK for me.

**Pull request merged**